trino/client.py

```diff
@@ -697,6 +697,9 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         rows = self._row_mapper.map(status.rows) if self._row_mapper else status.rows
 
         self._result = TrinoResult(self, rows)
+        # Execute should block until at least one row is received
+        while not self.finished and not self.cancelled and len(self._result._rows) == 0:
+            self._result._rows += self.fetch()
```

Comment:
You could add `rows` getter property and setter to `TrinoQuery` to not directly access `_rows` and then change it also in `columns` property.